<?xml version="1.0"?>
<launch>
  <arg name="robot" default="$(env ROBOT)"/>
  <arg name="pkg_moveit_config" default="$(find cob_moveit_config)"/>
  <arg name="config_path" default="$(arg pkg_moveit_config)/robots/$(arg robot)/moveit/config"/>

  <arg name="load_semantic_description" default="true"/>
  <arg name="load_planning_context" default="false"/>

  <arg name="verbose" default="false"/>
  
  <!-- Load SRDF -->
  <include file="$(arg pkg_moveit_config)/launch/upload_config.launch">
    <arg name="robot" value="$(arg robot)" />
    <arg name="config_path" value="$(arg config_path)" />
    <arg name="load_semantic_description" value="$(arg load_semantic_description)" />
    <arg name="load_planning_context" value="$(arg load_planning_context)" />
  </include>

  <node pkg="cob_collision_monitor" type="cob_collision_monitor_node" name="cob_collision_monitor_node" output="screen">
    <!--remap from="/cob_collision_monitor_node/state_is_valid" to="/safety_controller/state_is_valid"/-->  <!--only needed for the cob4 specific configuration-->
    <param name="verbose" value="$(arg verbose)"/>
    <param name="max_age" value="1.0"/>
    <param name="ground_link" value="base_link"/>
    <param name="ground_size" value="10.0"/>
    <param name="ground_height" value="-0.001"/>
  </node>

</launch>
